aeppl/transforms.py

```diff
@@ -205,6 +213,181 @@ def apply(self, fgraph: FunctionGraph):
+
+
+@_get_measurable_outputs.register(MeasurableTransform)
+def _get_measurable_outputs_Transform(op, node):
+    return [node.default_output()]
+
+
+@_logprob.register(MeasurableTransform)
+def measurable_transform_logprob(op: MeasurableTransform, values, *inputs, **kwargs):
+    """Compute the log-probability graph for a `MeasurabeTransform`."""
+    # TODO: Could other rewrites affect the order of inputs?
+    (value,) = values
+    other_inputs = list(inputs)
+    measurable_input = other_inputs.pop(op.measurable_input_idx)
+
+    # The value variable must still be back-transformed to be on the natural support of
+    # the respective measurable input.
+    backward_value = op.transform.backward(value, *other_inputs)
+    input_logprob = logprob(measurable_input, backward_value, **kwargs)
+
+    jacobian = op.transform.log_jac_det(value, *other_inputs)
+
+    return input_logprob + jacobian
+
+
+@local_optimizer([Elemwise])
+def find_measurable_transforms(
+    fgraph: FunctionGraph, node: Node
+) -> Optional[List[Node]]:
+    """Find measurable transformations from Elemwise operators with moret than one input.
+
+    Y = at.add(at.Normal(0, 1), 5)
+    """
+    scalar_op = node.op.scalar_op
+    if not isinstance(scalar_op, (Exp, Log, Add, Mul)):
+        return None
+
+    # Node was already converted
+    if isinstance(node.op, MeasurableVariable):
+        return None  # pragma: no cover
+
+    rv_map_feature: PreserveRVMappings = getattr(fgraph, "preserve_rv_mappings", None)
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    # Check that we have a single source of measurement
+    measurable_inputs = [
+        inp
+        for idx, inp in enumerate(node.inputs)
+        if inp.owner
+        and isinstance(inp.owner.op, MeasurableVariable)
+        and inp not in rv_map_feature.rv_values
+    ]
+
+    if len(measurable_inputs) != 1:
+        return None
+
+    measurable_input: TensorVariable = measurable_inputs[0]
+
+    # Do not apply rewrite to discrete variables
+    if measurable_input.type.dtype.startswith("int"):
+        return None
+
+    # Check that other inputs are not potentially measurable, in which case this rewrite
+    # would be invalid
+    other_inputs = tuple(inp for inp in node.inputs if inp is not measurable_input)
+    if any(
+        ancestor_node
+        for ancestor_node in walk_model(
+            other_inputs,
+            walk_past_rvs=False,
+            stop_at_vars=set(rv_map_feature.rv_values),
+        )
+        if (
+            ancestor_node.owner
+            and isinstance(ancestor_node.owner.op, MeasurableVariable)
+            and ancestor_node not in rv_map_feature.rv_values
+        )
+    ):
+        return None
+
+    # Make base_measure outputs unmeasurable
+    # This seems to be the only thing preventing nested rewrites from being erased
+    measurable_input = assign_custom_measurable_outputs(measurable_input.owner)
```

Comment:
@brandonwillard does this manipulation make sense now that rewrites apply regardless of being directly valued? I had to introduce it so that nested rewrites would work, otherwise the intermediate nodes would be replaced back to the original form. Clip, Cumsum, and Mixture all do this as well.

Reply:
Can you perform a check in `PreserveRVMappings` that prevents the relevant variables from being reverted when these transforms are performed?

Reply:
Not sure what such check would look like. How would we know if a variable in `PreserveRVMappings.measurable_conversions` is there because it was ultimately needed for deriving a valued variable or it was not needed.